core/src/main/java/com/oath/oak/InternalOakMap.java

```diff
@@ -775,24 +787,42 @@ boolean computeIfPresent(K key, Consumer<OakWriteBuffer> computer) {
             }
             return false;
         }
+
+        throw new RuntimeException("computeIfPresent failed: reached retry limit (1024).");
     }
 
     // used when value of a key was possibly moved and we try to search for the given key
     // through the OakMap again
-    EntrySet.LookUp refreshValuePosition(long keyReference) {
-        K deserializedKey = keySerializer.deserialize(getKeyByteBuffer(keyReference));
-        while (true) {
+    void refreshValuePosition(ThreadContext ctx) {
+        K deserializedKey = keySerializer.deserialize(ctx.key.getDataByteBuffer());
+
+        for (int i = 0; i < MAX_RETRIES; i++) {
             Chunk<K, V> c = findChunk(deserializedKey); // find chunk matching key
-            EntrySet.LookUp lookUp = c.lookUp(deserializedKey);
-            if (lookUp == null || lookUp.valueSlice == null) {
-                return null;
+            c.lookUp(ctx, deserializedKey);
+            if (!ctx.isValueValid()) {
+                return;
             }
 
-            if (updateVersionAfterLinking(c, lookUp)) {
+            if (updateVersionAfterLinking(c, ctx)) {
                 continue;
             }
-            return lookUp;
+            return;
+        }
+
+        throw new RuntimeException("refreshValuePosition failed: reached retry limit (1024).");
+    }
+
+    boolean refreshValuePosition(Slice keySlice, Slice valueSlice) {
```

Comment:
Please explain that valueSlice is actually used as output

Reply:
I updated the documentation.
Also, I updated the `refreshValuePosition(ctx)` method to return a boolean as well.